Automattic/pocket-casts-android
# Disable Private Feed Sharing

## Description
- This is to disable private podcast sharing and show a toast. See: pdeCcb-7Yf-p2
- Figma: HR2vcQrWcS0scsolKZBitg-fi-3567_9
- This change is behind: `share_podcast_private_not_available` feature flag
- See iOS PRs:
  - https://github.com/Automattic/pocket-casts-ios/pull/2582
  - https://github.com/Automattic/pocket-casts-ios/pull/2585


## Testing Instructions
- Note: I open the app with `debugProd` and used `https://pca.st/private/707a8bf0-9abf-013c-74fe-027201e7e97f` private feed
- Ensure you have the FF enabled

### Share In Podcast View
1. Open a private feed
2. Tap on the share button at the top
3. ✅ Ensure you can't share it
4. Play an episode from this private one
5. Open Now Playing
6. Tap on Share Button from shelf items list
7. ✅ Ensure you can't share it
8. Also, try to share the podcast from overflow menu
9. ✅ Ensure you can't share it

[Screen_recording_20241227_131449.webm](https://github.com/user-attachments/assets/e6f319e7-ac16-4e96-8dd8-a3336f3fb96a)

### Share from Podcasts Tab
1.  Follow private feed
2. Follow some public podcasts
3. Open the Podcasts tab
4. Tap on the 3 dots at the top to open the menu
5. Tap on Share Podcasts
6. ✅ Ensure you don't see the private one in the Select podcasts list

[podcasts.webm](https://github.com/user-attachments/assets/6cbc2979-d9b5-4225-a06e-bd6a04773dcf)

### Share Public podcasts
- Smoke test public podcasts and make sure you can share them


## Checklist
